src/transactions/TransactionFrame.cpp

```diff
@@ -826,22 +839,50 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
     }
     catch (std::exception& e)
     {
-        CLOG_ERROR(Tx, "Exception while applying operations ({}, {}): {}",
-                   xdr_to_string(getFullHash(), "fullHash"),
-                   xdr_to_string(getContentsHash(), "contentsHash"), e.what());
+        if (ledgerVersion.value() >= maxAssertLedgerVersion)
+        {
+            CLOG_ERROR(Tx, "Exception while applying operations ({}, {}): {}",
+                       xdr_to_string(getFullHash(), "fullHash"),
+                       xdr_to_string(getContentsHash(), "contentsHash"),
+                       e.what());
+        }
+        else
+        {
+            CLOG_INFO(
+                Tx,
+                "Exception occurred on outdated protocol version ({}) "
+                "while applying operations ({}, {}): {}",
+                ledgerVersion.value(), xdr_to_string(getFullHash(), "fullHash"),
+                xdr_to_string(getContentsHash(), "contentsHash"), e.what());
+        }
     }
     catch (...)
     {
-        CLOG_ERROR(Tx, "Unknown exception while applying operations ({}, {})",
-                   xdr_to_string(getFullHash(), "fullHash"),
-                   xdr_to_string(getContentsHash(), "contentsHash"));
+        if (ledgerVersion.value() >= maxAssertLedgerVersion)
+        {
+            CLOG_ERROR(Tx,
+                       "Unknown exception while applying operations ({}, {})",
+                       xdr_to_string(getFullHash(), "fullHash"),
+                       xdr_to_string(getContentsHash(), "contentsHash"));
+        }
+        else
+        {
+            CLOG_INFO(Tx,
+                      "Unknown exception on outdated protocol version ({}) "
+                      "while applying operations ({}, {})",
+                      ledgerVersion.value(),
+                      xdr_to_string(getFullHash(), "fullHash"),
+                      xdr_to_string(getContentsHash(), "contentsHash"));
+        }
     }
     // This is only reachable if an exception is thrown
     getResult().result.code(txINTERNAL_ERROR);
 
-    auto& internalErrorCounter = app.getMetrics().NewCounter(
-        {"ledger", "transaction", "internal-error"});
-    internalErrorCounter.inc();
+    // We only count internal error if the ledger number is and older version.
```

Comment:
Nitpick 1: typo "and" -> "an"

Nitpick 2: I think you mean ledger version here (ledger version is 18, ledger number is about 40M on production network)

Nitpick 3: I think you mean newer version